sync/class.jetpack-sync-module-users.php

```diff
@@ -54,6 +62,42 @@ public function init_listeners( $callable ) {
 		add_action( 'wp_masterbar_logout', $callable, 10, 0 );
 	}
 
+	public function delay_if_within_wp_insert_user() {
+		if ( ! Jetpack::is_function_in_backtrace( 'do_delayed_calls' ) ) {
```

Comment:
Consider collapsing to a single multi part conditional rather than two nested conditionals, e.g.
```
if ( ! Jetpack::is_function_in_backtrace( 'do_delayed_calls' ) && 
       Jetpack::is_function_in_backtrace( 'wp_insert_user' ) 
) {
     self::$calls_within_wp_insert_user[ current_action() ][] = func_get_args();

     return;
}
```